lms/djangoapps/instructor_task/tasks_helper/utils.py

```diff
@@ -45,3 +54,40 @@ def tracker_emit(report_name):
     Emits a 'report.requested' event for the given report.
     """
     tracker.emit(REPORT_REQUESTED_EVENT_NAME, {"report_type": report_name, })
+
+
+class SensitiveMessageOnReports(object):
+    """
+    Check how to add the disclaimer new row in the CSV,
+    due to reports are generated of different ways, each one handles their own way:
+        1. Using the CSV library directly.
+        2. Using upload_csv_to_report_store function which in turn
+           uses store_rows of DjangoStorageReportStore class where is built the CSV.
+    """
+
+    def __init__(self):
+        self.display_msg = configuration_helpers.get_value('DISPLAY_SENSITIVE_DATA_MSG_FOR_DOWNLOADS', settings.FEATURES.get('DISPLAY_SENSITIVE_DATA_MSG_FOR_DOWNLOADS', False))
+
+    def csv_direct(self, writer):
+        """
+        Writes the row immediately in the CSV.
+        """
+        if self.display_msg:
```

Comment:
To make it work I did this change
```
if self.should_msg_be_displayed:
            msg = self.process_message()
            encode = unicode(msg).encode('utf-8')
            writer.writerow([encode])

        return writer
```

Reply:
Done, did you try with all the reports? That's weird that have worked for you because there is another class that calls `should_msg_be_displayed` in lms/djangoapps/instructor_tasks/task_helpers/utils.py line 36:
```
if disclaimer.should_msg_be_displayed:
        # Append the new row above the headers.
        send_disclaimer = disclaimer.with_report_store()
        rows = [send_disclaimer] + rows
```

I have changed this name too.